### PR TITLE
Fix skinnable sounds playing more than once on skin change

### PR DESCRIPTION
**Pull request by peppy**
I thought I already fixed this one recently...

- [x] Depends on #10905 for ease of merging

### CODE REVIEW

**Comment by smoogipoo** [comment]
Keep in mind that a significant amount of this logic, including restart logic, was moved inside `PoolableSkinnableSample`. Is that implementation still fine? Or does it also need the `!Looping` check?

**Comment by peppy** [comment]
I didn't double check this after moving it across (only on master), probably needs to be there too.

**Comment by bdach** [comment]
FYI I'm holding off for a bit until the dependent PR gets merged before I review this.

**Comment by bdach** [comment]
Do you recall what the repro scenario for this is / whether this was in response to an open issue? I'm guessing it has to do with looping samples, although I'm not sure I understand why those shouldn't be restarted on skin change...

**Comment by peppy** [comment]
just smash the random skin hotkey while watching a replay. if it's still broken, the same sounds will play on already hit objects more than once.

**Comment by bdach** [comment]
Oh, I read the condition backwards, my bad. Will double-check master and this.

File: osu.Game/Skinning/PoolableSkinnableSample.cs
Comment (bdach) [inline osu.Game/Skinning/PoolableSkinnableSample.cs:105]: Upon actually checking, I actually don't think I originally misread and this should probably be

```suggestion
            if (wasPlaying && Looping)
```

right? The looping samples need to be restarted to continue playing, but the non-looping ones don't. It seems this logic can actually result in losing looping samples if I switch from skin A to B and then back to A again (i.e. during a spinner).

Comment (peppy) [reply osu.Game/Skinning/PoolableSkinnableSample.cs:105]: It does indeed look back to front.

Comment (bdach) [reply osu.Game/Skinning/PoolableSkinnableSample.cs:105]: In fact this actually might need to even be `wasPlaying || Looping` as I'm hitting some skins here that have `wasPlaying` set to apparently false despite being looping samples. I'm not yet 100% on what the cause of that would be, though.

Comment (bdach) [reply osu.Game/Skinning/PoolableSkinnableSample.cs:105]: Never mind, the case I'm hitting is caused by some frivolous audio handling framework-side without error checking.

**Comment by peppy** [comment]
Have double checked and the new behaviour sounds correct in gameplay.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**
